front/src/pods/student/student.component.tsx

```diff
@@ -11,25 +15,48 @@ interface Props {
 export const StudentComponent: React.FC<Props> = props => {
   const { room, log } = props;
 
+  const [isAutoScrollEnabled, setAutoScrollEnabled] = React.useState(true);
+
+  const textAreaRef = React.useRef(null);
+
+  React.useEffect(() => {
+    if (isAutoScrollEnabled && textAreaRef.current) {
+      textAreaRef.current.scrollTop = textAreaRef.current.scrollHeight;
+    }
+  }, [log]);
```

Comment:
Ok, Let's try to create reusable effect, psuedocode:

Option A:

```tsx
const useAutoScroll = () => {
  const [isAutoScrollEnabled, setIsAutoScrollEnabled] = React.useState(true);
  const textAreaRef = React.useRef(null);

  const doAutoScroll = () => {
     if(isAutoScrollEnabled && txtAreaRef.current) {
      textAreaRef.current.scrollTop = textAreaRef.current.scrollHeight;
    }
  }

  return {isAutoScrollEnabled, setIsAutoScrollEnabled, textAreaRef, doAutoScroll}
}
```

Usage

```tsx
const {isAutoScrollEnabled, setIsAutoScrollEnabled, textAreaRef, doAutoScroll} = useAutoScroll();

React.useEffect(() => {
  doAutoScroll();
}, [log]);
```

(...)

Not sure if we could some how pass 'log' (I think is not a good idea to keep it in the effect), and maybe we could create a textAreaWrapper that includes this funcionallity but maybe is just an overkill, I think just with this is good enough

Reply:
Ok, I was thinking of doing the same, where should I keep the hook?